datasets/sentinel-3/sentinel_3/sentinel_3_base.py

```diff
@@ -24,35 +32,148 @@
 }
 
 
+def recursive_round(coordinates: List[Any], precision: int) -> List[Any]:
```

Comment:
Style preference: this operates inplace and returns a value (the original list), which is a bit unusual. Functions mutating their input typically return None.

I'd prefer to either

- return a new list (and so don't mutate the input)
- operate in-place and don't return a value

As for choosing between those, slightly preference to returning a new list unless performance is an issue (which I imagine it isn't).

Reply:
Thanks. I'll take a look.

Reply:
`recursive_round` now returns a new list. And is simpler, to boot!